### PR TITLE
CI CD upgrade

### PR DESCRIPTION
**Pull request by Milstein**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,36 +4,37 @@ name: Build and Deploy Documentation using MkDocs
...
         with:
           python-version: ${{ matrix.python-version }}
-          architecture: x64
       - name: Install dependencies and setup git config
         run: |
           python -m pip install --upgrade pip
-          pip install -r requirements.txt
-          git config user.name 'github-actions[bot]' && git config user.email 'github-actions[bot]@users.noreply.github.com'
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          git config user.name 'github-actions[bot]' && git config user.email 'help@nerc.mghpcc.org'
```
Comment (aabaris) [inline .github/workflows/ci.yml:37]: are we sure we want to add help@nerc.mghpcc.org here?  what are the consequences?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,52 +1,63 @@
...
 
   - repo: https://github.com/Yelp/detect-secrets
-    rev: v1.1.0
+    rev: v1.5.0
     hooks:
-    - id: detect-secrets
-      args: ['--baseline', '.secrets.baseline']
-      exclude: .*/tests/.*
+      - id: detect-secrets
+        args: [--baseline, .secrets.baseline]
```
Comment (aabaris) [inline .pre-commit-config.yaml:49]: Usually I'd expect to see quotes in making an args list like this.  What's the guideline/reason that prompted to remove the quotes?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Milstein) [reply .github/workflows/ci.yml:37]: > Using the email address 'github-actions[bot]@users.noreply.github.com' for automated processes like GitHub Actions is a common practice. It ensures that automated commits are clearly associated with the bot account and don’t clutter your personal email inbox.

I will revert it back.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (Milstein) [reply .pre-commit-config.yaml:49]: > Best Practices:
> For simple argument values like file paths or numeric parameters, using args without quotes is common and considered best practice.

**Review by aabaris — APPROVED** [review]

**Pull request merged**